### PR TITLE
Switch npm to pnpm

### PR DESCRIPTION
**Pull request by Kevan-Y**
Fixes: [#2694](https://github.com/Seneca-CDOT/telescope/issues/2694)

Switch `npm` to `pnpm` engine

Local test steps:
1. `npm i -g pnpm`
2. `pnpm i`

### CODE REVIEW

File: .github/workflows/node-js-ci.yml [hunk 1]
```diff
@@ -34,21 +36,14 @@ jobs:
...
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v2
         with:
           node-version: ${{ matrix.node-version }}
-      - run: npm ci
-      - run: npm run prettier-check
-      - run: npm test
+          cache: 'pnpm'
+      - run: pnpm install
+      - run: pnpm prettier-check
```
Comment (humphd) [inline .github/workflows/node-js-ci.yml:48]: Let's move this to its own job. Make another job similar to the `eslint` one above, but have it run `prettier-check` vs. `eslint`.  It should only run on Ubuntu.

Let's rename this from `build` to `test`.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -70,7 +66,8 @@ jobs:
         with:
           node-version: '14.x'
```
Comment (humphd) [inline .github/workflows/release.yml:67]: Let's use LTS, which is 16

File: package.json [hunk 3]
```diff
@@ -9,11 +9,11 @@
     "coverage": "jest -c jest.config.js --collect-coverage",
     "eslint": "eslint --config .eslintrc.js \"**/*.js\"",
     "eslint-fix": "eslint --config .eslintrc.js \"**/*.js\" --fix",
-    "lint": "npm run eslint",
+    "lint": "pnpm eslint",
     "prettier": "prettier --write \"./**/*.{md,jsx,json,html,css,js,yml}\"",
     "prettier-check": "prettier --check \"./**/*.{md,jsx,json,html,css,js,yml}\"",
     "prepare": "husky install",
-    "pre-commit": "pretty-quick --staged && npm run lint"
+    "pre-commit": "pretty-quick --staged && pnpm run lint"
```
Comment (humphd) [inline package.json:16]: You don't need `run` with `pnpm`

Comment (humphd) [reply package.json:16]: Let's add a `preinstall` script like https://github.com/Seneca-CDOT/telescope/blob/master/package.json#L35

File: package.json [hunk 4]
```diff
@@ -39,7 +39,8 @@
     "pino-pretty": "7.5.0"
   },
   "engines": {
-    "node": ">=12.0.0"
+    "node": ">=12.0.0",
```
Comment (humphd) [inline package.json:42]: Let's bump to 14

**Review by humphd — CHANGES_REQUESTED** [review]

File: .github/workflows/node-js-ci.yml (hunk 1 above)
Comment (Kevan-Y) [reply .github/workflows/node-js-ci.yml:48]: Also, could we make `eslint` and `prettier-check` into its own file, like what telescope did?
https://github.com/Seneca-CDOT/telescope/blob/e1e546a58039d7bee642f5ee25fa55865f0a6524/.github/workflows/release.yml#L13-L18

Comment (humphd) [reply .github/workflows/node-js-ci.yml:48]: That's overkill.  I'd do them as separate jobs here, and they can run in parallel.

File: .github/workflows/node-js-ci.yml [hunk 5]
```diff
@@ -34,21 +53,13 @@ jobs:
...
       - name: Test
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v2
         with:
           node-version: ${{ matrix.node-version }}
-      - run: npm ci
-      - run: npm run prettier-check
-      - run: npm test
+          cache: 'pnpm'
+      - run: pnpm install
```
Comment (humphd) [inline .github/workflows/node-js-ci.yml:64]: Use the same multi-line syntax you do above:

```yaml
- run: |
    pnpm install
    pnpm test
```

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -19,25 +19,18 @@ jobs:
...
           node-version: ${{ matrix.node-version }}
-      - run: npm ci
-      - run: npm run prettier-check
-      - run: npm run eslint
-      - run: npm test
+          cache: 'pnpm'
+      - run: pnpm install
+      - run: pnpm prettier-check
+      - run: pnpm eslint
+      - run: pnpm test
```
Comment (humphd) [inline .github/workflows/release.yml:33]: Delete this entire `build` job.  We don't need it.

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -19,25 +19,18 @@ jobs:
...
 
   release:
     needs: [build]
```
Comment (humphd) [inline .github/workflows/release.yml:36]: Get rid of this

**Review by humphd — CHANGES_REQUESTED** [review]

**Review by humphd — APPROVED** [review]
Awesome

**Review by humphd — APPROVED** [review]

**Review by DukeManh — CHANGES_REQUESTED** [review]
You might also want to add `.pnpm-debug.log` in .gitignore like Telescope did.

**Comment by menghif** [comment]
> You might also want to add `.pnpm-debug.log` in .gitignore like Telescope did.

Other than that, it's great!

**Review by AmasiaNalbandian — APPROVED** [review]

**Pull request merged**